rsciio/jobin_yvon/api.py

```diff
@@ -0,0 +1,779 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2022 The HyperSpy developers
+#
+# This file is part of HyperSpy.
+#
+# HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HyperSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+from pathlib import Path
+import xml.etree.ElementTree as ET
+import numpy as np
+import logging
+from copy import deepcopy
+import importlib.util
+
+_logger = logging.getLogger(__name__)
+
+
+class JobinYvonXMLReader:
+    """Class to read Jobin Yvon .xml-files.
+
+    The file is read using xml.etree.ElementTree.
+    Each element can have the following attributes: attrib, tag, text.
+    Moreover, non-leaf-elements are iterable (iterate over child-nodes).
+    In this specific format, the tags not useful to extract information.
+    Instead, the "ID"-entry in attrib is used to identify the sort of information.
+    The IDs are consistent for the tested files.
+
+    Parameters
+    ----------
+    file_path: pathlib.Path
+        path to the to be read file
```

Comment:
```suggestion
        Path to the to be read file.
```
The common style is to use full sentences here. (see https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard )